webpack/components/SelectAllCheckbox/index.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Dropdown, DropdownToggle, DropdownToggleCheckbox,
+  DropdownItem } from '@patternfly/react-core';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { noop } from 'foremanReact/common/helpers';
+
+import { pluralize } from '../../utils/helpers';
+import './SelectAllCheckbox.scss';
+
+const SelectAllCheckbox = ({
+  selectAll,
+  selectNone,
+  selectPage,
+  selectedCount,
+  modelName,
+  modelNamePlural,
+  areAllRowsOnPageSelected,
+  areAllRowsSelected,
+}) => {
+  const [isSelectAllChecked, setSelectAllChecked] = useState(null);
+  const [isSelectAllDropdownOpen, setSelectAllDropdownOpen] = useState(false);
+
+  const onSelectAllCheckboxChange = () => {
+    if (isSelectAllChecked === false) {
```

Comment:
for false checks can we just bang it '! isSelectAllChecked"?

Reply:
Or in this case, you could do:
```suggestion
    if (isSelectAllChecked ) {
      return selectNone();
    }
     return selectPage();
```

Reply:
For this one I was explicit because it can have three values (`false`, `null`, `true`)

Reply:
if it is null you want it to hit select page?

Reply:
The checkbox states are:
Unchecked: false
Partially checked: null
Checked: true

The flow is:

All are selected (true) -> click -> selectNone()
Some are selected, or the page is selected (null) -> click -> selectNone()
None are selected (false) -> click -> selectPage()